### DISCUSSION TITLE
detail question - spli

### DISCUSSION THREAD

**Discussion by gilliM** [aspiaspace/EarthPT#4] [General]
Is it normal that you reference the same file for train_data and val_data at line 
https://github.com/aspiaspace/earthPT/blob/d8ceeb0159b144dca74e8736bad6ab9ee22d74bf/src/train.py#L142  ?

and also a view lines later, train_data array is reused ?

        if split == 'train':
            data = >>>>train_data<<< [np.random.randint(len(train_data))]
            ii = torch.randint(0, data.shape[0] - 10000, (batch_size,))
        else:
            data = >>>train_data<<< [np.random.randint(len(val_data))]

**Comment by Smith42** [comment]
Here we take the final 10000 timeseries in a preprocessed data array as our validation set (i.e. `torch.randint(data.shape[0] - 10000, data.shape[0], (batch_size,))` as below), and the rest are used as our training set (i.e. `torch.randint(0, data.shape[0] - 10000, (batch_size,))` as shown below and in your snippet).


```
def get_batch(split):
        if split == 'train':
            data = train_data[np.random.randint(len(train_data))]
            ii = torch.randint(0, data.shape[0] - 10000, (batch_size,))
        else:
            data = train_data[np.random.randint(len(val_data))]
            ii = torch.randint(data.shape[0] - 10000, data.shape[0], (batch_size,))
```

Does that clear things up? We are certainly not mixing the training and validation sets here!
